app/services/timebox.rb

```diff
@@ -1,13 +1,13 @@
 # Runs provided block within the time limit or else returns the result of the
-# fallback block. run may be called multiple times; the original time limit
-# being applied to all runs collectively.
+# optional fallback block. run may be called multiple times; the original time
+# limit being applied to all runs collectively.
 # Time limit may be an integer or a float.
 class Timebox
   def initialize(time_limit_seconds, start_time = Time.now.to_f)
     @deadline = start_time + time_limit_seconds
   end
 
-  def run(fallback_block)
+  def run(fallback_block = nil)
```

Comment:
The nil option allows it to be called with no fallback block - as from `AvailableSlotsController` - having the fallback block optional was suggested in Martyn's original prototype if you remember - I didn't think we'd need it until now.